swig/lightgbmlib.i

```diff
@@ -22,6 +24,26 @@
 					   int64_t* out_len) {
     char* dst = new char[buffer_len];
     int result = LGBM_BoosterSaveModelToString(handle, start_iteration, num_iteration, buffer_len, out_len, dst);
+    // Reallocate to use larger length
+    if (*out_len > buffer_len) {
+      delete [] dst;
+      int64_t realloc_len = *out_len;
+      dst = new char[realloc_len];
+      result = LGBM_BoosterSaveModelToString(handle, start_iteration, num_iteration, realloc_len, out_len, dst);
+    }
+    if (result != 0) {
+      return nullptr;
+    }
+    return dst;
+  }
+
+  char ** LGBM_BoosterGetEvalNamesSWIG(BoosterHandle handle,
+				       int eval_counts) {
+    char** dst = new char*[eval_counts];
+    for (int i = 0; i < eval_counts; i++) {
```

Comment:
`++i` for the consistency with cpp project's codebase?

Reply:
done!